rllib/agents/ppo/ppo.py

```diff
@@ -410,6 +418,86 @@ def get_default_policy_class(self, config: TrainerConfigDict) -> Type[Policy]:
         else:
             return PPOTFPolicy
 
+    @override(Trainer)
+    def training_iteration(self) -> ResultDict:
+        # Collects SampleBatches in parallel and synchronously
+        # from the Trainer's RolloutWorkers until we hit the
+        # configured `train_batch_size`.
+        sample_batches = []
+        num_env_steps = 0
+        num_agent_steps = 0
+        while (not self._by_agent_steps and num_env_steps < self.config[
+            "train_batch_size"]) or (
+            self._by_agent_steps and num_agent_steps < self.config["train_batch_size"]
+        ):
+            new_sample_batches = synchronous_parallel_sample(self.workers)
+            sample_batches.extend(new_sample_batches)
+            num_env_steps += sum(len(s) for s in new_sample_batches)
+            num_agent_steps += sum(s.agent_steps() for s in new_sample_batches)
+        self._counters[NUM_ENV_STEPS_SAMPLED] += num_env_steps
+        self._counters[NUM_AGENT_STEPS_SAMPLED] += num_agent_steps
+
+        # Combine all batches at once and treat as multi-agent from here on.
+        train_batch = SampleBatch.concat_samples(sample_batches)
+        train_batch = train_batch.as_multi_agent()
+
+        # Standardize `advantages` values in train_batch.
+        for policy_id, batch in train_batch.policy_batches.items():
+            if Postprocessing.ADVANTAGES not in batch:
+                raise KeyError(
+                    f"`{Postprocessing.ADVANTAGES}` not found in SampleBatch for "
+                    f"policy `{policy_id}`! Maybe this policy fails to add "
+                    f"`{Postprocessing.ADVANTAGES}` in its `postprocess_trajectory` "
+                    f"method? Or this policy is not meant to learn at all and you "
+                    "forgot to remove it from the list under `config."
+                    "multiagent.policies_to_train`."
+                )
+            batch[Postprocessing.ADVANTAGES] = standardized(
```

Comment:
why do we need this here, instead of in the policy's postprocessing function?

Reply:
I guess to standardize across the entire train batch, and not necessarily only across each (possibly truncated) episode trajectory.